src/TimeModeling/Modeling/distributed.jl

```diff
@@ -231,13 +231,15 @@ function reduce!(futures::Vector{_TFuture})
     # reminder
     R = M - m
 
+    pids = sort(collect(keys(futures)))
+    
     # Reduce the reminder
-    reduce_level!(futures, R)
+    reduce_level!(futures, R, m, pids)
 
     # Binary tree reduction
     for l = L:-1:1
         m = 2^(l-1)
-        reduce_level!(futures, m)
+        reduce_level!(futures, m, m, pids)
```

Comment:
Thanks for your branch `forfrancis` you also need to do
`reduce_level!(futures, m, m)`
at this line @mloubout